File: Benchmarks/Core/Components/SpriteComponent2D.cs
using Benchmarks.Core.Types;

namespace Benchmarks.Core.Components
{
    public struct SpriteComponent2D(int textureId, Rectangle sourceRectangle)
    {
        public int TextureId { get; set; } = textureId;
        public Rectangle SourceRectangle { get; set; } = sourceRectangle;
        public bool IsVisible { get; set; } = true;
        public bool Is2_5D { get; set; } = false;
    }
}

File: Benchmarks_FSharp/Core/Components/SpriteComponent2D.fs
namespace Benchmarks.Core.Components

open Benchmarks.Core.Types

[<Struct>]
type SpriteComponent2D =
    val mutable TextureId: int
    val mutable SourceRectangle: Rectangle
    val mutable IsVisible: bool
    val mutable Is2_5D: bool

    new(textureId: int, sourceRectangle: Rectangle) =
        { TextureId = textureId; SourceRectangle = sourceRectangle; IsVisible = true; Is2_5D = false }
